### PR TITLE
feat: disable run query button if query does not contain any changes

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/733

### Description:
disable button if query does not contain any changes

https://user-images.githubusercontent.com/962095/198999993-841729d8-83b8-47ec-837b-ea8b7d082e5c.mp4

### CODE REVIEW

**Comment by IrakliJani** [comment]
@TuringLovesDeathMetal should we add a tooltip when the button is disabled and why it is disabled?

**Comment by TuringLovesDeathMetal** [comment]
@IrakliJani ya, that would be awesome with a tooltip :) 

Maybe something like `You need to make some changes before running a query`

File: packages/frontend/src/components/RefreshButton.tsx [hunk 1]
```diff
@@ -10,53 +10,64 @@ export const RefreshButton = memo(() => {
...
+            content={
+                !hasUnfetchedChanges ? (
+                    'You need to make some changes before running a query'
+                ) : (
+                    <KeyCombo combo="cmd+enter" />
+                )
+            }
             position="bottom"
-            disabled={isDisabled || isLoadingResults}
+            disabled={isLoading || !isValidQuery}
```
Comment (ZeRego) [inline packages/frontend/src/components/RefreshButton.tsx:61]: Should this use `isButtonDisabled`?

File: packages/frontend/src/providers/ExplorerProvider.tsx [hunk 2]
```diff
@@ -1069,15 +1086,40 @@ export const ExplorerProvider: FC<{
...
+            });
+
+            return result;
+        } catch (e) {
+            console.error(e);
+        }
+    }, [mutateAsyncQuery, state.unsavedChartVersion]);
+
+    const hasUnfetchedChanges = useMemo(() => {
+        return !isEqual(
```
Comment (ZeRego) [inline packages/frontend/src/providers/ExplorerProvider.tsx:1108]: 👌 so good to have lodash now

**Review by ZeRego — APPROVED** [review]
🐑 it!

**Pull request merged**